### DISCUSSION TITLE
Visitor location not getting trakced.

### DISCUSSION THREAD

**Discussion by Zonalds** [vinceanalytics/vince#22] [General]
Hey Vince, 

Great work you have done here. I can't find a way to create an issue so I am using the discussion tab instead. 

I noticed the visitor location isn't getting tracked. The countries tab isn't recording any data. 

Any fix for this?

**Comment by gernest** [comment]
Can you maybe share a screen shot of the dashboard  maybe I can spot something of interest ?

We process location based on IP, so as long as an IP is received we can correctly derive location info.

**Comment by Zonalds** [comment]
Sure, here:

I just installed it a couple hours ago, not sure if it needs time?


![image](https://github.com/user-attachments/assets/77063162-8870-4d74-937a-0176e3fb3df1)

**Reply by gernest** [reply]
Looks like no valid IP was sent to vince.

**Reply by Zonalds** [reply]
> Looks like no valid IP was sent to vince.

How can I fix that?

**Reply by gernest** [reply]
There is little I can do here since it is not a vince problem.  I will explain how we find IP based on a event sent to vince and it iis your responsibly to ensure correct IP reaches vince to be able to see location data.

note that we use free db-ip geoip database, so it is possible the IP is not here, but highly unlikely at least you should be able to see countries.

We select IP in this order

- `x-vince-ip` header
- `cf-connecting-ip` header
- `b-forwarded-for` header
- `X-Real-IP` header
- `X-Forwarded-For` header
- `X-Client-IP` header
- Remote IP of the client connection

Whichever is set first will be used.

**Reply by Zonalds** [reply]
Thank you,

I updated the Nginx config and everything worked alright.
